src/main/java/com/google/sps/database/DatabaseImpl.java

```diff
@@ -0,0 +1,278 @@
+/*
+ * Implementation for DatabaseImpl class
+ *  Current public functions at a glance:
+ *  test()
+ *  addVideo()
+ *  addClause()
+ *  addClauses()
+ *  addMetadata()
+ *  getAllKeywords()
+ *  getTimesForKeyword()
+ *  videoInDb()
+ * 
+ */
+
+package com.google.sps.database;
+
+import com.google.appengine.api.datastore.*;
+import java.util.*;
+
+public class DatabaseImpl implements DatabaseInterface {
+
+    
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    private static final String V_KIND = "video";
+    private static final String M_KIND = "metadata";
+    private static final String C_KIND = "caption";
+    private static final String T_KIND = "timestamps";
+
+    //================================================================================
+    // General Test Function (not used for official testing)
+    //================================================================================
+    
+    public void test() {
```

Comment:
Add it to a separate integration test file if needed. It is quite confusing to have a test function in this class